### PR TITLE
update package to use mpi-serial, add extra module info

### PR DESCRIPTION
**Pull request by jedwards4b**
Add support for mpi-serial builds.  (depends on https://github.com/spack/spack/pull/33150)
Add module information used by some dependent libraries.

### CODE REVIEW

File: var/spack/repos/builtin/packages/parallelio/package.py [hunk 1]
```diff
@@ -29,24 +29,26 @@ class Parallelio(CMakePackage):
     variant(
         "fortran", default=True, description="enable fortran interface (requires netcdf fortran)"
     )
+    variant("mpi", default=True, description="Use mpi to build, otherwise use mpi-serial")
 
-    depends_on("mpi")
-    depends_on("netcdf-c +mpi", type="link")
+    depends_on("mpi", when="+mpi")
+    depends_on("mpi-serial", when="-mpi")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/parallelio/package.py]: We use `~` almost everywhere, can you change that throughout the recipe?
```suggestion
    depends_on("mpi-serial", when="~mpi")
```

**Review by alalazo — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/parallelio/package.py (hunk 1 above)
Comment (jedwards4b) [reply var/spack/repos/builtin/packages/parallelio/package.py]: done

**Comment by bernhardkaindl** [comment]
@alalazo: @jedwards4b said he applied your suggested change. Please approve.

**Review by alalazo — APPROVED** [review]

File: var/spack/repos/builtin/packages/parallelio/package.py [hunk 2]
```diff
@@ -29,24 +29,26 @@ class Parallelio(CMakePackage):
...
         "fortran", default=True, description="enable fortran interface (requires netcdf fortran)"
     )
+    variant("mpi", default=True, description="Use mpi to build, otherwise use mpi-serial")
 
-    depends_on("mpi")
-    depends_on("netcdf-c +mpi", type="link")
+    depends_on("mpi", when="+mpi")
+    depends_on("mpi-serial", when="~mpi")
+    depends_on("netcdf-c +mpi", type="link", when="+mpi")
+    depends_on("netcdf-c -mpi", type="link", when="~mpi")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/parallelio/package.py]:
```suggestion
    depends_on("netcdf-c ~mpi", type="link", when="~mpi")
```

**Review by alalazo — CHANGES_REQUESTED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**